reframe/frontend/executors/policies.py

```diff
@@ -411,12 +371,8 @@ def _finalize_task(self, task):
 
     def _failall(self, cause):
         '''Mark all tests as failures'''
-        for partname in self._running_tasks.keys():
-            try:
-                while True:
-                    self._running_tasks[partname].pop().abort(cause)
-            except IndexError:
-                pass
+        for task in sum(self._running_tasks.values(), []):
```

Comment:
```suggestion
        for task in itertools.chain(*self._running_tasks.values()):
```

And if `abort()` removes it from the list, then wrap the above inside a `list()`.